libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/Parser.kt

```diff
@@ -256,14 +216,71 @@ internal class Parser(src: String, private val withSourceLocation: Boolean, val
 
     val arguments = parseArguments(const = false)
     val directives = parseDirectives(const = false)
+    var nullability: GQLNullability? = null
+
+    if (allowClientControlledNullability) {
+      nullability = parseNullability()
```

Comment:
Shouldn't this be before `parseDirectives`?

Reply:
Good catch 🙏